### ISSUE TITLE
Support search resolve for better repo: filter recognition in 1 click insight creation flow.

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#22005]
[Origin thread about this problem ](https://github.com/sourcegraph/sourcegraph/pull/21943#discussion_r649310260)

Now we use regexp for processing values from `repo:` filters but this logic is not reliable in many cases 

- `repo:^github.com/sourcegraph/sourcegraph-`
- `repo:^github.com/sourcegraph/.*\.js$`
- `repo:^github.com/sourcegraph/sourcegraph-(sourcegraph|about)$`

In order to support all possible cases, we should use a search resolver and use search results instead of runtime regexp processed values from URL. 

Now we support only `repo:` filter with simple straightforward value like
- `repo:^github.com/sourcegraph/about$`
- `repo:^github.com/sourcegraph/about$|^github.com/sourcegraph/about$`
- `repo: github.com/sourcegraph/about`

**Comment by vovakulikov** [comment]
TLDR of previous converstiona about what is the UX about run search resolve 

@vovakulikov 
>add this properly since search resolve is an async operation I think we should add some other UI state to repositories field in a creation UI and put some explanation comment below the field like - trying resolve repo: search query or something like this. 

@felixfbecker 
>I would have thought of this happening on the search page, e.g. using the <LoaderButton> component we have. When you click the button, it resolves the repo filter to a list, then redirects to the create page with the resolved repos in the query parameter.

@vovakulikov 
>Hmm. I would like to have this button just a simple link that user can open it in the current window or in a new tab immediately and then can see a creation UI with maybe loading state

>With the loader button on the search page, I see a couple of difficulties to clarify the workflow. What if a user clicked create insight button and then while our repo resolve was going user changed the query string in the search bar? What if the same scenario happened if a user clicked with the command key should we delay opening a new tab and cancel in case if a user changed the search query?

>Also, code monitoring works as a simple link probably we want to have consistency here as well



@Joelkw @AlicjaSuska FYI

**Comment by felixfbecker** [comment]
(I'm fine either way)

**Comment by Joelkw** [comment]
@vovakulikov what do we think the "average upper bound" on loading time for this would be? If >1 second, I think I have a preference for doing the loading on a new tab / the create page rather than the search page – less likely to get distracted by your existing page, and you know/can see that something is definitely happening (and we can even put an explanatory message). But this is a not a "strong" preference for 3.29 – if this is too complicated to handle then we can return to that in a month.

**Comment by vovakulikov** [comment]
@Joelkw It's hard to say due to all users have different internet connections but I would say that the average time is more than 1s. So I also prefer to do loading on a new tab. @AlicjaSuska we probably need to determine the right visual/loading state for that. We  can just add loader with some description while loading/fetching all needed repositories is going, or show the creation UI and add loading state right next repositories field, wdyt?
